storage.go

```diff
@@ -473,6 +473,7 @@ func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairSt
 	input := &s3.GetObjectInput{
 		Bucket: aws.String(s.name),
 		Key:    aws.String(rp),
+		Range:  aws.String(fmt.Sprintf("bytes=%d-%d", opt.Offset, opt.Offset+opt.Size)),
```

Comment:
`Range` is `[a, b]`, maybe we need to use `opt.Offset+opt.Size-1` for end?

Reply:
BTW, we need to check offset and size before we use them.

- What if the offset is missing?
- What if the size is missing?

Reply:
> `Range` is `[a, b]`, maybe we need to use `opt.Offset+opt.Size-1` for end?
Got

Reply:
> BTW, we need to check offset and size before we use them.
> 
> * What if the offset is missing?
> * What if the size is missing?

Both offset and size are not pointer types, so we have to check whether size is greater than 0